test/func_ds3_share_open_access.py

```diff
@@ -282,6 +288,52 @@ def _run_all(self, fstype, tests, testtype):
         ret = carry ? -1 : 0;
         break;
       }
+
+      case SETATT: {
+        /*
+          DOS 2+ - CHMOD - SET FILE ATTRIBUTES
+        */
+
+        const char *fname = FN1 "." FE1;
+        unsigned int oattr, nattr;
+        int rc;
+
+        rc = _dos_getfileattr(fname, &oattr);
+        if (rc != 0) {
+          printf("FAIL:('%s', '%s', '%s')[secondary getattr1 failed(%d)]\n",
+            argv[2], argv[3], argv[4], rc);
+          return -1;
+        }
+        if (oattr & _A_HIDDEN) {
+          printf("FAIL:('%s', '%s', '%s')[secondary getattr1 invalid(0x%02x)]\n",
+            argv[2], argv[3], argv[4], oattr);
+          return -1;
+        }
+
+        rc = _dos_setfileattr(fname, oattr | _A_HIDDEN);
+        if (rc != 0) {
+          printf("INFO:('%s', '%s', '%s')[secondary setattr failed(%d)]\n",
+            argv[2], argv[3], argv[4], rc);
+        } else {
+          int rc2;
+
+          printf("INFO:('%s', '%s', '%s')[secondary setattr success]\n",
```

Comment:
Shouldn't that fail?

Reply:
No, in the spirit of printing all possible info as we go along, the failures are processed at the end.

Reply:
So in the context of our trying to set attr whilst the file is open, the setattr call failed, but the overall test is success because that is the expected behaviour from MS-DOS 6.22.

Reply:
And if the setattr call succeeded we want to continue to read back what we set so we can print it as info for debugging.